# Project Overview
This retrospective study examines cardiovascular (CV) risk assessment and toxicity monitoring in 239 advanced prostate cancer patients treated with novel hormonal therapies (abiraterone, enzalutamide, apalutamide, and darolutamide). Using data extracted from UCI’s Epic electronic medical record system (64) variables per patient), the project aims to evaluate how consistently providers assess baseline CV risk factors, identify which factors predict post-treatment CV toxicities, and determine patterns in antihypertensive use before and after therapy initiation. The ultimate goal is to leverage data analytics and AI to identify high-risk patients, develop clear infographics to visualize findings, and potentially implement an automated referral pathway to cardio-oncology within Epic.

## Import Required Libraries and Load Raw Data

In [5]:
import pandas as pd
import os

excel_file = os.path.join('..', 'data', 'raw', 'Cardio_onc_prostate_students.xlsx')
df = pd.read_excel(excel_file)

## Remove Metadata Row

The second row of the dataset contained encoding information describing categorical variable mappings (1-Yes, 2-No, 3-Not Recorded). As this row doesn't represent patient-level observations, it is removed.

### Before Removal

In [6]:
df.head(3)

,Unique Patient ID,Ethnicity,"NHT Prior authorization date (Abiraterone, Enzalutamide, Apalutamide, Darolutamide)","Start date of NHT (Abiraterone, Enzalutamide, Apalutamide, Darolutamide)",BMI (when they started NHT),Specific NHT used,Age at NHT start,"Start date of ADT (Lupron, Firmagon, Orgovyx, or Bicalutamide)",Specific ADT agent used,Hx of smoking?,...,Saw cards after NHT start?,Referred to cards post NHT start by oncology?,Documented counseling about diet by oncology w/i 3 mos of NHT start?,Documented counseling about exercise by oncology w/i 3 mos of NHT start?,Order for echo w/i 3 mos of NHT start?,ECG w/i 3 month of NHT start?,QTc,Actively seen by non-oncology providers?,"has PCP (1:No, 2: at UCI 3: Outside).1",Prescribing Provider
0,NaN,( 1- Caucasian; 2- Hispanic; 3- Asian; 4- Blac...,NaN,NaT,NaN,NaN,NaN,NaN,"(Lupron, Firmagon, Orgovyx, or Bicalutamide)","(0- No, 1- Former, but quit within one year of...",...,"(1- yes, 2- no)","(1- yes, 2- no)","(1- yes, 2- no)","(1- yes, 2- no)","(1- yes, 2- no)","(1- yes, 2- no)",NaN,"0: no, 1: cardio, 2: PCP/geri, 3: endo",NaN,NaN
1,1.0,NaN,2022-01-09 00:00:00,NaT,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"MEI, ERIKA A"
2,2.0,NaN,2022-01-11 00:00:00,NaT,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"MEI, ERIKA A"


### After Removal

In [7]:
df = df.drop(index=0).reset_index(drop=True)
df.head(3)

,Unique Patient ID,Ethnicity,"NHT Prior authorization date (Abiraterone, Enzalutamide, Apalutamide, Darolutamide)","Start date of NHT (Abiraterone, Enzalutamide, Apalutamide, Darolutamide)",BMI (when they started NHT),Specific NHT used,Age at NHT start,"Start date of ADT (Lupron, Firmagon, Orgovyx, or Bicalutamide)",Specific ADT agent used,Hx of smoking?,...,Saw cards after NHT start?,Referred to cards post NHT start by oncology?,Documented counseling about diet by oncology w/i 3 mos of NHT start?,Documented counseling about exercise by oncology w/i 3 mos of NHT start?,Order for echo w/i 3 mos of NHT start?,ECG w/i 3 month of NHT start?,QTc,Actively seen by non-oncology providers?,"has PCP (1:No, 2: at UCI 3: Outside).1",Prescribing Provider
0,1.0,NaN,2022-01-09 00:00:00,NaT,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"MEI, ERIKA A"
1,2.0,NaN,2022-01-11 00:00:00,NaT,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"MEI, ERIKA A"
2,3.0,1,2022-01-12 00:00:00,2022-02-01,25.99,Darolutamide,73.0,2018-01-26 00:00:00,Lupron,1,...,2,2,2,2,2,NaN,NaN,NaN,2.0,"MEI, ERIKA A"


## Clean and Standardize Column Names

In [8]:
# Strip them, lower case, and replace spaces with underscores
df.columns = df.columns.str.strip().str.lower().str.replace(' ', '_')
df.head(3)

,unique_patient_id,ethnicity,"nht_prior_authorization_date_(abiraterone,_enzalutamide,_apalutamide,_darolutamide)","start_date_of_nht_(abiraterone,_enzalutamide,_apalutamide,_darolutamide)",bmi_(when_they_started_nht),specific_nht_used,age_at_nht_start,"start_date_of_adt_(lupron,_firmagon,_orgovyx,_or_bicalutamide)",specific_adt_agent_used,hx_of_smoking?,...,saw_cards_after_nht_start?,referred_to_cards_post_nht_start_by_oncology?,documented_counseling_about_diet_by_oncology_w/i_3_mos_of_nht_start?,documented_counseling_about_exercise_by_oncology_w/i_3_mos_of_nht_start?,order_for_echo_w/i_3_mos_of_nht_start?,ecg_w/i_3_month_of_nht_start?,qtc,actively_seen_by_non-oncology_providers?,"has_pcp_(1:no,_2:_at_uci_3:_outside).1",prescribing_provider
0,1.0,NaN,2022-01-09 00:00:00,NaT,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"MEI, ERIKA A"
1,2.0,NaN,2022-01-11 00:00:00,NaT,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"MEI, ERIKA A"
2,3.0,1,2022-01-12 00:00:00,2022-02-01,25.99,Darolutamide,73.0,2018-01-26 00:00:00,Lupron,1,...,2,2,2,2,2,NaN,NaN,NaN,2.0,"MEI, ERIKA A"


As we can see, there are still parantheses that contain more metadata like: "has PCP (1:No, 2: at UCI, 3: Outside)". Other column names also have metadata, but is not enclosed within parantheses such as "On Beta-blocker prior to NHT? Acebutolol, Atenolol, Bisoprolol, Carvedilol, Metoprolol, Nadolol, Nebivolol, Propranolol, Betaxolol, Esmolol, Nebivolol, Carteolol, Penbutolol, Pindolol, Labetalol, Sotalol"

In [10]:
# Remove everything in the parantheses in the column names, 
# remove metadata outside of the parantheses, and replace spaces with underscores
# Manual inspection may be required !!!!

## Inspect Data Types and Structure

In [11]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 240 entries, 0 to 239
Data columns (total 64 columns):
 #   Column                                                                                                                                                                                                                          Non-Null Count  Dtype         
---  ------                                                                                                                                                                                                                          --------------  -----         
 0   unique_patient_id                                                                                                                                                                                                               239 non-null    float64       
 1   ethnicity                                                                                                                

240 Total Rows and 64 Total Columns of many different types. Potentialy column type conversion could be good for future modeling. Further inspection is required.

In [ ]:
# Conversion

## Missing Data

In [17]:
missing_summary = pd.DataFrame({
    'null_count': df.isnull().sum(),
    'null_percentage': df.isnull().mean() * 100
}).sort_values(by='null_count', ascending=False)
print(missing_summary)

                                                    null_count  \
date_of_carotid_art_disease                                236   
date_of_pvd/pad                                            236   
ef                                                         234   
date_of_hf                                                 234   
date_of_cva                                                234   
...                                                        ...   
specific_nht_used                                            7   
nht_prior_authorization_date_(abiraterone,_enza...           3   
10-year_ascvd_risk                                           3   
prescribing_provider                                         3   
unique_patient_id                                            1   

                                                    null_percentage  
date_of_carotid_art_disease                               98.333333  
date_of_pvd/pad                                           98.333333

In [18]:
high_null_cols = missing_summary[missing_summary['null_percentage'] > 50]
print(high_null_cols)

                                          null_count  null_percentage
date_of_carotid_art_disease                      236        98.333333
date_of_pvd/pad                                  236        98.333333
ef                                               234        97.500000
date_of_hf                                       234        97.500000
date_of_cva                                      234        97.500000
date_of_mi_orstent                               233        97.083333
date_of_arrhythmia                               229        95.416667
date_of_cad                                      222        92.500000
a1c                                              219        91.250000
qtc                                              207        86.250000
10-yr_risk_score_%                               199        82.916667
ldl                                              190        79.166667
hdl                                              190        79.166667
tc                  

Variables with more than 60% missing data will be evaluated for exclusion.

## Feature Viability

Specified feature categories for questions of interest! 

## Summary and Next Steps

In [ ]:
# Save preprocessed dataframe to csv
#df.to_csv('preprocessed_cardio_onc.csv', index=False)

After initial preprocessing, 